### PR TITLE
Added payOutEstimate view function returns uint256[6]

### PR DESCRIPTION
**Pull request by mrk-hub**
Added function payoutEstimate() returns(uint256[6])
list of returns represents to [document](https://colab.research.google.com/drive/1RBybbYGcC2oxZIW1egCDIM5i67hb27OE?usp=sharing)
where element 0 is a loss
element 1-5 match P1-P5.

### CODE REVIEW

File: src/DegenGambit.sol [hunk 1]
```diff
@@ -605,6 +605,21 @@ contract DegenGambit is ERC20, ReentrancyGuard {
         }
     }
 
+    // Payout Estimate function to easily display current payouts estimate at time of function call
+    function payoutEstimate(
+    ) external view returns (uint256[6] memory payouts) {
+        payouts[0] = 0;
```
Comment (zomglings) [inline src/DegenGambit.sol:611]: What's the point of this?

File: bindings/DegenGambit/DegenGambit.go
Comment (zomglings) [inline bindings/DegenGambit/DegenGambit.go:1]: Why did you delete this file?

Comment (zomglings) [reply bindings/DegenGambit/DegenGambit.go:1]: You deleted a lot of files.

File: src/DegenGambit.sol [hunk 2]
```diff
@@ -605,6 +605,21 @@ contract DegenGambit is ERC20, ReentrancyGuard {
         }
     }
 
+    // Payout Estimate function to easily display current payouts estimate at time of function call
+    function payoutEstimate(
```
Comment (zomglings) [inline src/DegenGambit.sol:609]: I want to see a test or multiple tests which test the values of this `view` against those of the `payout` method.

Actually, in the integration guide we should perhaps just tell users to use the `payout` method?

Or maybe we should just return something like:

```solidity
return (payout(0,0,0), payout(1,1,1), payout(15, 16, 15), payout(15, 16, 17), payout(17, 17, 17));
```

(My indices may be incorrect, but the idea stands).

**Review by zomglings — CHANGES_REQUESTED** [review]
Some feedback.

File: src/DegenGambit.sol (hunk 1 above)
Comment (mrk-hub) [reply src/DegenGambit.sol:611]: I figured it gave a proper list of payouts based on the document for degen gambit.
For others using non-official websites this is still an option for payout.
Also this allows for the array indexes to follow the document P1, P2, ... , P5 for a more direct understanding and comparison.

File: test/DegenGambit.t.sol [hunk 3]
```diff
@@ -926,4 +926,50 @@ contract DegenGambitTest is Test {
...
+        assertEq(payoutEstimate[0], payout);
+
+        payout = degenGambit.payout(17, 16, 18);
+        assertEq(payoutEstimate[3], payout);
+
+        payout = degenGambit.payout(16, 2, 16);
+        assertEq(payoutEstimate[0], payout);
+
+        payout = degenGambit.payout(16, 17, 16);
+        assertEq(payoutEstimate[4], payout);
```
Comment (mrk-hub) [inline test/DegenGambit.t.sol:973]: This test will fail until Payout(left,center,right) is updated to account for it.

Comment (zomglings) [reply test/DegenGambit.t.sol:982]: This is resolved now?

File: src/DegenGambit.sol (hunk 1 above)
Comment (zomglings) [reply src/DegenGambit.sol]: We decided to remove the `0` payout.

File: src/DegenGambit.sol [hunk 4]
```diff
@@ -601,6 +601,23 @@ contract DegenGambit is ERC20, ReentrancyGuard {
         }
     }
 
+    // Payout Estimate function to easily display current payouts estimate at time of function call
+    function payoutEstimate()
```
Comment (zomglings) [inline src/DegenGambit.sol:605]: This name is a bit confusing.

We are using `payout` to refer to the outcome of a single reel configuration (with `left`, `center`, and `right` inputs).

Maybe we should call this function `prizeEstimates`?

Or just `prizes` would be better IMO.

**Review by zomglings — CHANGES_REQUESTED** [review]
Left a comment re: name of method. I suggest `payoutEstimate` -> `prizes`.

Also, you should add documentation of the new view method to the integration guide (see `docs/DegenGambitIntegrationGuide.md`).

File: src/DegenGambit.sol (hunk 4 above)
Comment (mrk-hub) [reply src/DegenGambit.sol:605]: Yes, I'll make that change.

File: test/DegenGambit.t.sol (hunk 3 above)
Comment (mrk-hub) [reply test/DegenGambit.t.sol:982]: Yes this is reseolved

**Review by zomglings — APPROVED** [review]
lgtm

Code changes look good. Please add `prizes` documentation to integration guide before merging.
